### ISSUE TITLE
Batch updating entries with m2m field results in duplicates 

### ISSUE THREAD

**Issue by julianreimer** [directus/directus#15034]
### Describe the Bug

Batch updating entries which include a many to many field results in duplicates if options are added to it. 

### To Reproduce

Create a new collection called _Items_ and another collection named _Options_. Add a new Input field to _Options_ called _Name_. Create at least 2 new entries in _Options_ with _Name_ set to something unique. Create a new Many to Many field in _Items_ and set _Options_ as the related collection. 

Create a few entries in _Items_ and select some _Options_. You should not be able to add _Options_ you have already selected. 

Go to the overview table of _Items_ and select all entries. Open the batch editor and choose one or more _Options_ and hit save. The entries of _Items_ should contain the _Options_ which were added individually and in addition to that all the _Options_ which were selected in the batch editor. 

https://user-images.githubusercontent.com/10420902/184177359-ad634579-8a3c-4ab6-aecf-11191d8fb72f.mov

### What version of Directus are you using?

9.15.1

### What version of Node.js are you using?

Default Directus Docker Image

### What database are you using?

Default Directus Docker Image 

### What browser are you using?

Chrome

### How are you deploying Directus?

Docker

**Comment by rijkvanzanten** [comment]
This is technically working as intended, but I agree that the behavior is unexpected and should be considered a bug 👍🏻

**Comment by Nitwel** [comment]
I'm wondering how we would go ahead and fix that. As to fix it in the api we would have to either introduce a batch mode to relational interfaces or we would have to fix that inside the api.

**Comment by rijkvanzanten** [comment]
@Nitwel We might be able to add a "delete everything" flag next to the create calls for the batch mode in the app side, something like:

```
{
  create: [{ title: 'example' }],
  delete: ['*']
}
```

**Comment by Nitwel** [comment]
I don't think that is a solution either as there could be information on the junction table that would get lost in the process. My intuition would be to add ones that dont exist yet and delete the ones, that existed, but are not selected in the batch mode.

**Comment by rijkvanzanten** [comment]
> My intuition would be to add ones that dont exist yet and delete the ones, that existed, but are not selected in the batch mode.

Hmm yeah, though the goal here is to set all items in the batch to the same result, so maybe we'd have to clear out all existing linked junction items after all 🤔

**Comment by domdus** [comment]
Dealing with the duplicates issue right now.
We are distributing content (articles, videos, images) to different "services" (customers). So the content collections have a m2m to `services` collection. During batch you don't know what items have a service assigned already, so "batch-adding" it again lead to duplicates.
I would vote for a  `add ones that dont exist yet` as a fix.

Also I see how a `delete everything`  or  a `delete the ones, that existed, but are not selected in the batch mode` would solve the duplication totally, but here a concern:

A main use case we have is -> launching a new service:
That means hundrets of content items get selected to be batch-updated to add a new (created) `service`.
In this case a `delete everything` would basically drop items that may have been already assigned to 1-n services/customers.

**Comment by bcartier** [comment]
It seems to me that the existing wording and functionality already sets the expectation of batch **editing** not batch adding. The way to works for other multi-value fields like checkboxes etc is that it completely replaces all values (checked unchecked not-checked) with the newly specified selections (checked or not checked. Not selecting a relationship in the Bulk Edit is just as meaningful as selecting one. 

Choosing to Bulk **edit** relationships and having the system leave existing relationships as-is but adding only newly selected is a bug. It only implements half of the intent of the UI.  

If the use case exists to "Batch add" (as opposed to "Batch edit"), then that should be accomplished with a new feature or perhaps a Flow or something.

**Comment by rijkvanzanten** [comment]
I agree. The expectation is set that the interface will update every selected item to the _same_ value, rather than modifying the relations on top of what's already there.

**Comment by rijkvanzanten** [comment]
Linear: [ENG-210](https://linear.app/directus/issue/ENG-210/batch-updating-entries-with-m2m-field-results-in-duplicates)

**Comment by kajal-g-7span** [comment]
Is there any update on this issue? I am still encountering this problem with many-to-many relationships, where new items are being created as duplicates during bulk updates.
